admin/src/react-components/service-editor.js

```diff
@@ -224,14 +225,45 @@ class ConfigurationEditor extends Component {
 
   renderLongTextInput(path, descriptor, currentValue) {
     const displayPath = path.join(" > ");
+
+    function isValidJSON(s) {
+      try {
+        JSON.parse(s);
+        return true;
+      } catch (error) {
+        console.error(error);
+        return false;
+      }
+    }
+
     return (
       <TextField
         key={displayPath}
         id={displayPath}
         label={descriptor.name || displayPath}
         inputProps={{ maxLength: 4096 }}
         value={currentValue || ""}
-        onChange={ev => this.onChange(path, ev.target.value)}
+        onChange={ev => {
```

Comment:
Maybe not for this PR, but I dont know if I like that the popup will keep popping up every time you type within an invalid json object. 

How does it feel when you type in the field? -- if it feels fine, then this works.

Maybe having the textfield highlight in red with the initial pop up of invalid json. Which the red will go away afterward. You can do this with a `style={isValidJson ? {} : {border: "2px solid red"}}` (or `styles` I've forgotten). Disregard if you think the "feeling"  of typing in invalid json is fine.

Reply:
You could also check if the warningMessage is already set with the matching descriptor name/path before setting this.